src/lib/util/ipfs.ts/ipfs.ts

```diff
@@ -0,0 +1,15 @@
+/**
+ * Pulls the IPFS hash from an IPFS url
+ * @param url IPFS url to get hash from
+ * @returns IPFS hash from url
+ */
+export const getHashFromIPFSUrl = (url: string) => {
+	const regex = /Qm(\w{44})[/\w]*/;
+
+	if (regex.test(url)) {
+		const matches = url.match(regex) as string[];
+		return matches[0];
+	}
+
+	return '';
```

Comment:
Unsure what this function should do if the `regex.test` returns false. Returning empty string feels weird - anything using this function will have to check `!== ''` or `.length > 1`. Maybe returning `undefined` is better here.